Business Card Ui development

Kevin-Guillen opened the pull request:
Business Card UI Project 


<img width="224" alt="Business_Card" src="https://user-images.githubusercontent.com/84336422/120726284-188a4300-c4ae-11eb-8914-4b1f31398db3.png">

Kevin-Guillen committed:
Business Card Ui development

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                  color: Colors.black12,
+                  child: ListTile(
+                    leading: Container(
+                      padding: const EdgeInsets.fromLTRB(0.0, 8.0, 0.0, 0.0),
```
In here you can wrap the Container with a Padding widget
And for the padding you can use EdgeInsets.only(top: 8.0,), because you only modify the top value of Padding.

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                  color: Constants.cardsBackgroundColor,
+                  child: ListTile(
+                    leading: Container(
+                      padding: const EdgeInsets.fromLTRB(0.0, 8.0, 0.0, 0.0),
```
In here you can do the same

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                    ),
+                  ),
+                ),
+                SizedBox(height: 10.0),
```
missed a trailing comma here and the value can be added to the constants file

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                    shape: BoxShape.circle,
+                    boxShadow: [
+                      BoxShadow(
+                          blurRadius: 35,
```
This value can be added to the constants file

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                Text(
+                  'Flutter Developer Trainee',
+                  style: TextStyle(
+                    fontSize: 20.0,
```
this value could go in the constants file

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                      letterSpacing: 1.0,
+                    )),
+                SizedBox(
+                  height: 10.0,
```
this value could go in the constants file

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                  ),
+                ),
+                SizedBox(
+                  height: 45.0,
```
this value could go in the constants file

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                  ),
+                  child: CircleAvatar(
+                    backgroundImage: AssetImage('assets/images/me.jpg'),
+                    radius: 100,
```
this value could go in the constants file

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,128 @@
+                    letterSpacing: 1.0,
+                  ),
+                ),
+                SizedBox(
+                  height: 80.0,
+                  width: 300.0,
+                  child: Divider(
+                    thickness: 2.0,
+                    color: Colors.blueAccent.shade100,
+                  ),
+                ),
```
You can remove the SizedBox and follow the example of this page https://api.flutter.dev/flutter/material/Divider-class.html, to use the full configuration of the Divider widget

MarcosCGlobant reviewed (changes requested):
The SafeArea implementation requested for the exercise is missing.
You can remove the colors from the Constants file, they are already constants.
And some changes that I left commented.

Kevin-Guillen committed:
- Wrapped two containers with a padding widget
- Added the missing constants
- Replace a SizedBox with the divider widget
- Added the App icon.

marianosalvetti reviewed (approved):
Looks good

FranciscoGaraventa commented on lib/main.dart:
```diff
@@ -1,113 +1,142 @@
+                  image: AssetImage("assets/images/cardbackground.png"),
+                  fit: BoxFit.fill)),
+          child: SafeArea(
+            top: true,
+            bottom: true,
+            left: true,
+            right: true,
```
This is unnecessary. SafeArea by default has this parameters as true.

Kevin-Guillen replied:
fixed

FranciscoGaraventa commented on lib/main.dart:
```diff
@@ -1,113 +1,142 @@
+      debugShowCheckedModeBanner: false,
+      title: 'Business Card',
+      home: Scaffold(
+        body: Container(
+          decoration: BoxDecoration(
+              image: DecorationImage(
+                  image: AssetImage("assets/images/cardbackground.png"),
+                  fit: BoxFit.fill)),
```
Try to move this Container inside SafeArea. If you have problem to refactor this changes use a Stack as child of SafeArea.

FranciscoGaraventa replied:
You missed a trailing comma in line 20 and 21.

eis6aer replied:
Another good practice woulld be to create an assets class to store references to the given assets, let me know if this is clear but I think @MarcosCGlobant or @FranciscoGaraventa would know what I mean

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/main.dart:
```diff
@@ -1,113 +1,142 @@
+                      BoxShadow(
+                          blurRadius: Constants.avatarBlurShadow,
+                          color: Constants.avatarShadowColor,
+                          spreadRadius: 1)
```
You missed a trailing comma here.

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/main.dart:
```diff
@@ -1,113 +1,142 @@
+                      fontWeight: FontWeight.bold,
+                      color: Constants.personNameTextColor,
+                      letterSpacing: Constants.letterSpacing,
+                    )),
```
You missed a trailing comma here.

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/main.dart:
```diff
@@ -1,113 +1,142 @@
+                    ],
+                  ),
+                  child: CircleAvatar(
+                    backgroundImage: AssetImage('assets/images/me.jpg'),
```
You missed a trailing comma here.

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/main.dart:
```diff
@@ -1,113 +1,142 @@
+                    ),
+                  ),
+                ),
+                SizedBox(height: Constants.phoneAndEmailSizeBoxHeight),
```
You missed a trailing comma here.

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/utils/constants.dart:
```diff
@@ -0,0 +1,23 @@
+  static const double emailFontSize = 20.0;
+  static const double avatarBlurShadow = 35.0;
+
+}
```
You missed an extra line that is need it by convention. Use CTRL+ALT+L to solved it.

Kevin-Guillen replied:
fixed

MarcosCGlobant reviewed (commented):
Approve after Fran's comments are solved

Kevin-Guillen committed:
- Added missing trailing commas
- Solved the extra line in the constants file
- Deleted the unnecessary SafeArea parameters from SafeArea
- Put the avatar container inside the SafeArea

eis6aer commented on lib/main.dart:
```diff
@@ -1,113 +1,151 @@
+                      letterSpacing: Constants.letterSpacing,
+                    ),
+                  ),
+                  Column(
```
Is this column needed ? you could omit the column and just use the divider instead right ?

Kevin-Guillen replied:
done

eis6aer commented on lib/main.dart:
```diff
@@ -1,113 +1,151 @@
+                    child: ListTile(
+                      leading: Padding(
+                        padding: EdgeInsets.only(
+                          top: 8.0,
```
This could be a constant like mediumPadding ?

Kevin-Guillen replied:
fixed

eis6aer commented on lib/main.dart:
```diff
@@ -1,113 +1,151 @@
+                      ),
+                      subtitle: Text(
+                        "Cell Phone",
+                        style: TextStyle(
```
For the future, you could create TextStyles to be used all around your app, instead of having to create each of them every time, for example, `titleTextStyle` which will have the standard style for the texts which are headers or titles ?

Kevin-Guillen replied:
done

eis6aer commented on pubspec.yaml:
```diff
@@ -44,6 +50,9 @@ flutter:
   # the material Icons class.
   uses-material-design: true
 
+  assets:
```
You can decide whether to specify each file on a new line, or just make the reference to `assets/images/` and then all the contents will be added to assets, just a reminder 👍

Kevin-Guillen replied:
done

eis6aer reviewed (approved):
LGTM

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,151 @@
+      home: Scaffold(
+        body: Container(
+          child: SafeArea(
+            top: false,
```
why did you put the SafeArea top in false

Kevin-Guillen replied:
fixed

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,151 @@
+                    image: AssetImage(
+                      "assets/images/cardbackground.png",
+                    ),
+                    fit: BoxFit.fill),
```
missed a trailing comma here

Kevin-Guillen replied:
done

MarcosCGlobant reviewed (commented):
a missing comma and a question

Kevin-Guillen committed:
- Deleted the unnecessary column
- Created a styles.dart file where i put the styles of the contact info
- Instead of specifying each image that i've used , i just referenced to assets/images

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,134 @@
+      debugShowCheckedModeBanner: false,
+      title: 'Business Card',
+      home: Scaffold(
+        body: Container(
+          color: Colors.purple.shade300,
```
you can use this color value on the Scaffold attribute backgroundColor, and then remove this Container, it should have the same effect

Kevin-Guillen replied:
done

MarcosCGlobant commented on lib/utils/constants.dart:
```diff
@@ -0,0 +1,18 @@
+  // contactInfoTextStyle
+  static const mailAndPhoneNumberFontSize = 20.0;
+  static const letterSpacing = 1.0;
+  static const contactInfoTextColor = Colors.white;
```
This constant is not been used, you should remove it, also try not to make constants out of constants

Kevin-Guillen replied:
done

MarcosCGlobant commented on lib/main.dart:
```diff
@@ -1,113 +1,134 @@
+            child: Container(
+              decoration: BoxDecoration(
+                image: DecorationImage(
+                  image: AssetImage('assets/images/cardbackground.png',),
```
Do Ctrl + Alt + L to fix the indentation problem on the file

Kevin-Guillen replied:
done

MarcosCGlobant reviewed (commented):
indentation and constants problems

Kevin-Guillen committed:
- Removed unnused constant
- Fixed identation problem
- Removed unnesesary container
- added missing constants
- Replaced background image widget from decoration to stack

Kevin-Guillen committed:
- Removed unnused constant
- Fixed identation problem
- Removed unnesesary container

MarcosCGlobant reviewed (approved):
LGTM
